### PR TITLE
Feature/3269 draft idea hub posts

### PR DESCRIPTION
**Pull request by ivankruchkoff**
## Summary

Addresses issue #3269 

## Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: assets/js/modules/idea-hub/datastore/draft-ideas.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+		};
+	},
+	argsToParams: ( idea ) => {
+		return { idea };
+	},
+	validateParams: ( { idea } ) => {
+		invariant( isPlainObject( idea ), 'idea must be an object.' );
+		invariant( typeof idea.name === 'string', 'idea.name must be a string.' );
+		invariant( typeof idea.text === 'string', 'idea.text must be a string.' );
+		invariant( Array.isArray( idea.topics ), 'idea.topics must be a string.' );
```
Comment (tofumatt) [inline assets/js/modules/idea-hub/datastore/draft-ideas.js:50]:
```suggestion
		invariant( Array.isArray( idea.topics ), 'idea.topics must be an array.' );
```

File: assets/js/modules/idea-hub/datastore/draft-ideas.test.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+	afterEach( () => {
+		unsubscribeFromAll( registry );
+	} );
+
+	afterAll( () => {
+		API.setUsingCache( true );
+	} );
+
+	describe( 'actions', () => {
+		describe( 'createIdeaDraftPost', () => {
```
Comment (tofumatt) [inline assets/js/modules/idea-hub/datastore/draft-ideas.test.js:52]: Can you add a test for the error state as well? Something like https://github.com/google/site-kit-wp/blob/81130720be9540c4c6e9d60e348c4409654e2016/assets/js/googlesitekit/data/create-fetch-store.test.js#L208-L225

**Review by tofumatt — CHANGES_REQUESTED** [review]
Looks good to me. There's one typo in an `invariant` check (copypasta I think? 😄 ) and I think the JS tests could use another one for error state/handling, but after that this is good-to-go 👍🏻

File: assets/js/modules/idea-hub/datastore/draft-ideas.test.js (hunk 2 above)
Comment (ivankruchkoff) [reply assets/js/modules/idea-hub/datastore/draft-ideas.test.js:52]: Added in https://github.com/google/site-kit-wp/pull/3369/commits/4dc1379b59cd3f1e48a4faf92f1bc629ae5a12d4

**Review by tofumatt — APPROVED** [review]
Looks great! 👍🏻

**Pull request merged**